server/src/main/java/io/unitycatalog/server/service/credential/CredentialOperations.java

```diff
@@ -50,6 +44,20 @@ public GenerateTemporaryTableCredentialResponse vendCredentialForTable(TableInfo
     return vendCredential(credentialContext).toTableCredentialResponse();
   }
 
+  public GenerateTemporaryModelVersionCredentialResponse vendCredentialForModelVersion(ModelVersionInfo modelVersionInfo, Set<CredentialContext.Privilege> privileges) {
+    String mvStorageLocation = modelVersionInfo.getStorageLocation();
+    if (mvStorageLocation == null || mvStorageLocation.isEmpty()) {
+      throw new BaseException(ErrorCode.FAILED_PRECONDITION, "Model version storage location not found.");
+    }
+
+    URI storageLocationUri = URI.create(mvStorageLocation);
```

Comment:
Will this work if mvStorageLocation is a relativePath ? we do set relative paths in our test cases/sample database.

Reply:
I don't think I currently set relative paths for this in models or model versions.

Reply:
We are setting relative paths in the quick start h2 db for external tables so that it resolves correctly in everyone's setup to `etc/data` relative to the project root and quick start commands work out of box. That's one use case.

Reply:
think the bigger question would be why the client (or someone) is asking for temporary credentials for local filesystem/paths

Reply:
There is a precedence in CLI : https://github.com/ravivj-db/unitycatalog/blob/main/examples/cli/src/main/java/io/unitycatalog/cli/TableCli.java#L172. But I agree, the client can check prior to sending the credential request for path scheme(if path is available prior). 

However, in this particular case, we first fetch the ModelVersionInfo from model repository while serving the temporary credential request and use it's storageLocation field. If that's relative (which sometimes happens for pre-stored entities) `URI.create(mvStorageLocation)` will fail. In that case something like 
`Path path = Paths.get(mvStorageLocation);
URI uri = path.toUri();` 
will make more sense. 

We can also embed this conversion in `getModelVersionInfo` function itself during conversion from `DAO` to `Info` object.

Reply:
I think CLI needs updating and possibly not ask for creds if its a local path (it also needs to update since it's expecting AwsCredentials only...). If anything in the credentials space, should we throw error or return empty creds if we see a local file path (whether we use Paths to then create the URI or not)?

Reply:
https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/service/credential/CredentialOperations.java#L68 .. We are returning empty credentials here. Let's check once with UC and keep the behaviour consistent.

Reply:
So I don't think it makes sense to put relative paths in storage location.  I believe we treat the storage location as a URI in DB UC.  I don't think you can create a relative path file URI.

Reply:
It was just a hack for making the quickstart work in everyone's own environment. We can discuss it later when we are putting together the quickstart commands for models.